// naive-ui/es/number-animation/src/NumberAnimation.js
import { defineComponent, computed, onMounted, ref, watchEffect } from 'vue';
import { round } from 'lodash-es';
import { tween } from './utils';
const numberAnimationProps = {
    to: {
        type: Number,
        default: 0
    },
    precision: {
        type: Number,
        default: 0
    },
    showSeparator: Boolean,
    from: { type: Number, default: 0 },
    active: {
        type: Boolean,
        default: true
    },
    duration: {
        type: Number,
        default: 2000
    }
};
export default defineComponent({
    name: 'NumberAnimation',
    props: numberAnimationProps,
    setup(props) {
        const { duration } = props;
        const displayedValueRef = ref(props.from);
        let animating = false;
        const onUpdate = (currentValue) => {
            displayedValueRef.value = currentValue;
        };
        const onFinish = () => {
            displayedValueRef.value = props.to;
            animating = false;
        };
        const animate = (from = props.from, to = props.to) => {
            animating = true;
            displayedValueRef.value = props.from;
            if (from !== to) {
                tween({
                    from,
                    to,
                    duration: duration,
                    onUpdate,
                    onFinish
                });
            }
        };
        const formattedValueRef = computed(() => {
            const formatted = round(displayedValueRef.value, props.precision).toFixed(props.precision);
            const splitValue = formatted.split('.');
            const integer = props.showSeparator
                ? Number(splitValue[0]).toLocaleString('en-US')
                : splitValue[0];
            const decimal = splitValue[1];
            return {
                integer,
                decimal
            };
        });
        function play() {
            if (animating)
                return;
            animate();
        }
        onMounted(() => {
            watchEffect(() => {
                if (props.active)
                    animate();
            });
        });
        const exposedMethods = { play };
        return Object.assign({ formattedValue: formattedValueRef }, exposedMethods);
    },
    render() {
        const { formattedValue: { integer, decimal } } = this;
        return [integer, decimal ? '.' : null, decimal];
    }
});
